exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -229,6 +288,67 @@ func (pq *persistentQueue[T]) unrefClient(ctx context.Context) error {
 	return nil
 }
 
+// marshalQueueMetadata serializes the queue metadata.
+func marshalQueueMetadata(meta *queueMetadata, buf []byte) ([]byte, error) {
+	buf = buf[:0] // Clear the buffer before use if it's being reused
+	buf = binary.LittleEndian.AppendUint64(buf, meta.ReadIndex)
+	buf = binary.LittleEndian.AppendUint64(buf, meta.WriteIndex)
+	//nolint:gosec
+	buf = binary.LittleEndian.AppendUint64(buf, uint64(meta.QueueSize))
+
+	cdiLen := len(meta.CurrentlyDispatchedItems)
+	//nolint:gosec
+	buf = binary.LittleEndian.AppendUint32(buf, uint32(cdiLen))
+	for _, item := range meta.CurrentlyDispatchedItems {
+		buf = binary.LittleEndian.AppendUint64(buf, item)
+	}
+	return buf, nil
+}
+
+// unmarshalQueueMetadata deserializes the queue metadata.
+func unmarshalQueueMetadata(data []byte) (*queueMetadata, error) {
+	if data == nil {
+		return nil, errValueNotSet
+	}
+	// Minimum size: ReadIndex(8) + WriteIndex(8) + QueueSize(8) + CDILen(4) = 28
+	if len(data) < 28 {
+		return nil, fmt.Errorf("queue metadata too short: %d bytes, expected at least 28", len(data))
+	}
+
+	meta := &queueMetadata{}
+	offset := 0
+	meta.ReadIndex = binary.LittleEndian.Uint64(data[offset : offset+8])
+	offset += 8
+	meta.WriteIndex = binary.LittleEndian.Uint64(data[offset : offset+8])
+	offset += 8
+	queueSize := binary.LittleEndian.Uint64(data[offset : offset+8])
+	//nolint:gosec
+	meta.QueueSize = int64(queueSize)
+	offset += 8
```

Comment:
See my comments in the original PR, you need a size type, because that may change between restarts.